### PR TITLE
Add SpatialConstant

### PR DESCRIPTION
**Pull request by isuruf**
Partially fixes https://github.com/inducer/pytential/issues/116

### CODE REVIEW

File: sumpy/symbolic.py [hunk 1]
```diff
@@ -243,6 +244,33 @@ def find_power_of(base, prod):
     return result[power]
 
 
+class SpatialConstant(prim.Variable):
```
Comment (inducer) [inline sumpy/symbolic.py:247]: I'd prefer if this had its own `mapper_method`, instead of piggybacking off of `map_variable` and using `isinstance`.

Comment (isuruf) [reply sumpy/symbolic.py:247]: What do you think about moving this to `pymbolic`? If we don't move this `pymbolic`, issue is that, I have to override a lot of mappers of pymbolic in both sumpy and pytential with `map_spatial_constant=map_variable`.

File: sumpy/symbolic.py [hunk 2]
```diff
@@ -243,6 +244,30 @@ def find_power_of(base, prod):
...
```
Comment (inducer) [inline sumpy/symbolic.py:247]: Hook this into the documentation and write a docstring explaining the motivation.

Comment (isuruf) [reply sumpy/symbolic.py:255]: Done in 8e7435a. Ready for review

**Pull request merged**